enhancements/machine-config/admin-defined-reboot-policy.md

```diff
@@ -0,0 +1,264 @@
+---
+title: admin-defined-reboot-policy
+authors:
+  - "@yuqi-zhang"
+reviewers:
+  - TBD
+approvers:
+  - TBD
+api-approvers:
+  - TBD
+creation-date: yyyy-mm-dd
+last-updated: yyyy-mm-dd
+tracking-link:
+  - TBD
+see-also:
+  - https://issues.redhat.com/browse/OCPSTRAT-380
+  - https://issues.redhat.com/browse/MCO-507
+replaces:
+superseded-by:
+---
+
+# Admin Defined Reboot Policy
```

Comment:
At an overall thing I think we don't want to hardcode the term "reboot" here because some changes today are still disruptive in terms of draining, but don't actually involve a reboot right?  (Hm, the code flow in the MCO today is a bit confusing on this; I'd have expected `postConfigChangeActionDrain` but it's special cased)

Certainly in bootc I plan at some point to support systemd userspace-only restarts too which will act differently from reboots.

And then this whole enhancement is really about enabling special casing specific changes which are not reboots...

So maybe "Admin defined Host Change Apply Policy" ?

Reply:
I picked "reboots" since that's how most people view it :) I was considering the alternative of "Admin defined disruption policy" instead but `disruption` has multiple meanings

The title I can certainly change to be more descriptive, the naming of the api object (or other alternatives) is definitely more important